src/main/java/com/github/cameltooling/lsp/internal/instancemodel/propertiesfile/CamelPropertyValueInstance.java

```diff
@@ -48,8 +49,12 @@ public CamelPropertyValueInstance(String camelPropertyFileValue, CamelPropertyKe
 	}
 
 	public CompletableFuture<List<CompletionItem>> getCompletions(Position position, CompletableFuture<CamelCatalog> camelCatalog) {
-		if (new CamelKafkaUtil().isCamelURIForKafka(key.getCamelPropertyKey())) {
+		String propertyKey = key.getCamelPropertyKey();
+		if (new CamelKafkaUtil().isCamelURIForKafka(propertyKey)) {
 			return new CamelEndpointCompletionProcessor(textDocumentItem, camelCatalog).getCompletions(position);
+		} else if (new CamelKafkaUtil().isConnectorClassForCamelKafkaConnector(propertyKey)) {
+			String startFilter = camelPropertyValue.substring(0, position.getCharacter() - key.getEndposition() -1);
```

Comment:
this logic is also used in the else case. I am wondering if we should extract it into a utils function for better sharing.